### ISSUE TITLE
Exceptions are swallowed

### ISSUE THREAD

**Issue by venkatd** [zino-app/graphql-flutter#177]
Related to #140 

In most of the `try ... catch` clauses in the code, we are catching all exceptions. This results in errors unrelated to the library (such as network errors) getting swallowed which makes things very difficult to debug.

For example:
https://github.com/zino-app/graphql-flutter/blob/f7d50caf4e4999761ac8c128097ff90b3e19bb99/lib/src/widgets/query.dart#L90

https://github.com/zino-app/graphql-flutter/blob/f7d50caf4e4999761ac8c128097ff90b3e19bb99/lib/src/widgets/query.dart#L107

The solution would be to catch on specific exceptions. We do this in our own code for example:

```
  _login(BuildContext context) async {
    try {
      await auth.signIn();

      Navigator.of(context).pushReplacement(
        new MaterialPageRoute(
          builder: (context) {
            return DashboardContainer();
          },
        ),
      );
    } on GoogleSignInFailedError catch (error) {
      _showLoginFailed(context, error);
    } on TurtleSignInFailedError catch (error) {
      _showLoginFailed(context, error);
    }
  }
```

If there is any other exception besides what we explicitly catch, it will properly get raised so a developer can debug easily.

**Comment by HofmannZ** [comment]
I think we could solve this by creating our own kind of exceptions so that we can handle them cleaner internally.

**Comment by truongsinh** [comment]
closed by #193
